go-app/report/analyzeLog.go

```diff
@@ -65,48 +65,49 @@ type FullDetails struct {
 }
 
-var (
-	FullLogDetails = FullDetails{}
-	CfgFile        = Config{}
-)
-
-func AnalyseLog(w http.ResponseWriter, r *http.Request, project_id string, region_id string) error {
+func AnalyseLog(w http.ResponseWriter, r *http.Request, project_id string, region_id string, fullLogDetails *FullDetails, cfgFile *Config) error {
 	fContent, fName, cfgName, bucket, err := uploadLogFile(w, r, project_id, region_id)
 	if err != nil {
 		return err
 	}
-	FullLogDetails.Analysis_details = AnalysisDetails{}
+	fullLogDetails.Analysis_details = AnalysisDetails{}
 	//Set the selected platform
-	FullLogDetails.Analysis_details.Platform = bucket
-	cfgFile, err := extractConfig(cfgName, bucket)
+	fullLogDetails.Analysis_details.Platform = bucket
+	err = extractConfig(cfgName, bucket, cfgFile)
 	if err != nil {
 		return err
 	}
-	CfgFile = *cfgFile
-	FullLogDetails.GroupedIssues = make(map[string]GroupedStruct)
-	FullLogDetails.NonGroupedIssues = make(map[string]map[string]bool)
-	FullLogDetails.Analysis_details.FileName = *fName
-	FullLogDetails.Analysis_details.RawLog = fContent
-	getSpecProcessLogs(cfgFile, fContent)
+	fullLogDetails.GroupedIssues = make(map[string]GroupedStruct)
+	fullLogDetails.NonGroupedIssues = make(map[string]map[string]bool)
+	fullLogDetails.Analysis_details.FileName = *fName
+	fullLogDetails.Analysis_details.RawLog = fContent
+	fullLogDetails.Analysis_details.SpecificProcess = make(map[string]string)
+	spec_proc_map := fullLogDetails.Analysis_details.SpecificProcess
+	setSpecProcessLogs(cfgFile, fContent, spec_proc_map)
 	//Fill the header with general fields
 	headerMap := map[string]bool{"Issue": true, "Number": true, "Details": true, "Timestamp": true, "LogLevel": true}
 	for field, _ := range cfgFile.IssuesGeneralFields.OtherFields {
 		headerMap[field] = true
 	}
-	getIssueDetails(cfgFile, fContent, headerMap)
-	FullLogDetails.Analysis_details.OrderedIssues = sortIssue(cfgFile)
-	FullLogDetails.Analysis_details.Header = fillHeader(headerMap)
+	fullLogDetails.Analysis_details.Issues = make(map[string]map[string]string)
+	issues_map := fullLogDetails.Analysis_details.Issues
+	grp_issues := fullLogDetails.GroupedIssues
+	ngrp_issues := fullLogDetails.NonGroupedIssues
+	getIssueDetails(cfgFile, fContent, headerMap, issues_map, spec_proc_map, grp_issues, ngrp_issues)
+	fullLogDetails.Analysis_details.OrderedIssues = make([]string, len(cfgFile.Issues), len(cfgFile.Issues))
+	sortIssue(cfgFile, fullLogDetails.Analysis_details.OrderedIssues)
+	fullLogDetails.Analysis_details.Header = fillHeader(headerMap)
 	return nil
 }
-func sortIssue(cfgFile *Config) []string {
-	issues := make([]string, 0, len(cfgFile.Issues))
+func sortIssue(cfgFile *Config, issues []string) {
+	index := 0
 	for k := range cfgFile.Issues {
-		issues = append(issues, k)
+		issues[index] = k
+		index++
 	}
 	sort.Slice(issues, func(i, j int) bool {
 		return cfgFile.Priority[issues[i]] > cfgFile.Priority[issues[j]]
 	})
-	return issues
```

Comment:
I prefer this method returning the list of issues rather than mutating the object that was passed in. I generally try to avoid modifying objects that are passed as parameters unless it's explicitly documented.

Reply:
Maybe I wrong and I didn't really understand what I read about golang. I read that in Golang it is better to create an object and pass it as reference to a function (and the function will populate or mutate that) than creating the object in the function and returning that object, because in this case , it is more likely for the object to be stored in heap memory and that is not good. 
I used that logic in all of my code and would need to change that if I am wrong